homeworks/week6/VuNguyen.java

```diff
@@ -0,0 +1,73 @@
+package homeworks.week6;
+
+// Given a string S and a string T,
+// find the minimum window in S which will contain all the characters in T
+// Input: S = "xyyzyzyx" T = "xyz”
+// Output: "zyx"
+
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class VuNguyen {
+    public static void main(String[] args) {
+        System.out.println(getShortestSubstringOfAllCharacters("xyyzyzyx", "xyz"));
+        // "zyx"
+
+        System.out.println(getShortestSubstringOfAllCharacters("xyxz", "xyz"));
+        // "yxz"
+        System.out.println(getShortestSubstringOfAllCharacters("xyz", "xyz"));
+        // "xyz"
+        System.out.println(getShortestSubstringOfAllCharacters("xyz", "y"));
+        // y
+        System.out.println(getShortestSubstringOfAllCharacters("xyz", "z"));
+        // z
+    }
+
+    // assume arr alway contain all the characters in T for sure
+    // assume T alway contain distinct character
+    // time complexity: O(n)
+    // space complexity: O(n) for set
```

Comment:
You have not define what is `n`.

The correct answer is O(M) where M = T.length.